### PR TITLE
Make subgroups unique to group chats

### LINKED ISSUE #20: Pull In-memory DBs into conf test file.
Move 
```
@pytest.fixture
def session():
```
into a shared `conftest` file to prevent code duplication

### LINKED ISSUE #24: Make subgroups unique to group chats
Different group chats can have the same subgroup name. We need to make sure subgroup names are only unique to the group chat

### PR DESCRIPTION
**Pull request by AshciR**
closes #24 
closes #20

### CODE REVIEW

File: src/shout_subgroup/models.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ class SubgroupModel(Base):
     created_at = Column(DateTime, default=datetime.now(timezone.utc))
     updated_at = Column(DateTime)
     users = relationship("UserModel", secondary=users_subgroups_join_table, backref="subgroups")
+    table_args = (UniqueConstraint('group_chat_id', 'name', name='_group_chat_id_name_uc'))
```
Comment (AshciR) [inline src/shout_subgroup/models.py:46]: Focus: Added this table constraint. group chat id and (subgroup)name together has to be unique for each row

File: src/shout_subgroup/repository.py [hunk 2]
```diff
@@ -4,20 +4,22 @@
...
 
 
-async def find_all_users_in_subgroup(db: Session, subgroup_name: str) -> list[Type[UserModel]]:
+async def find_all_users_in_subgroup(db: Session, group_chat_id: int, subgroup_name: str) -> list[Type[UserModel]]:
     users = (
         db.query(UserModel)
         .join(users_subgroups_join_table)
         .join(SubgroupModel)
-        .filter(SubgroupModel.name == subgroup_name)
+        .filter(SubgroupModel.name == subgroup_name, SubgroupModel.group_chat_id == group_chat_id)
```
Comment (AshciR) [inline src/shout_subgroup/repository.py:16]: Focus: Finds all the users that belong to the subgroup FOR that group chat

File: src/shout_subgroup/shout.py [hunk 3]
```diff
@@ -1,22 +1,37 @@
...
+from shout_subgroup.repository import find_all_users_in_group_chat, find_all_users_in_subgroup, \
+    find_group_chat_by_telegram_group_chat_id
+from shout_subgroup.utils import is_group_chat
 
 
-async def shout_subgroup_members(db: Session, subgroup_name: str) -> str:
+async def shout_subgroup_members(db: Session, telegram_chat_id: int, subgroup_name: str) -> str:
 
-    all_members = await find_all_users_in_subgroup(db, subgroup_name)
+    if not await is_group_chat(telegram_chat_id):
```
Comment (AshciR) [inline src/shout_subgroup/shout.py:18]: Boyscout: Added Guard clauses for `/shout` command

File: tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+from sqlalchemy.orm import sessionmaker
+
+from shout_subgroup.models import Base
+
+# Create an SQLite in-memory database and a session factory
+engine = create_engine('sqlite:///:memory:', echo=True)
+SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+
+@pytest.fixture
```
Comment (AshciR) [inline tests/conftest.py:12]: Refactor: Moved the test db setup into a common file

File: tests/test_create_subgroup.py [hunk 5]
```diff
@@ -1,45 +1,10 @@
 from unittest.mock import Mock
 
 import pytest
-from sqlalchemy import create_engine
-from sqlalchemy.orm import sessionmaker, Session
+from sqlalchemy.orm import Session
 
-from shout_subgroup.models import Base, UserModel, GroupChatModel
 from shout_subgroup.create_subgroup import create_subgroup
-
```
Comment (AshciR) [inline tests/test_create_subgroup.py:9]: Logic no longer needed b/c of the new helper functions and db being moved into common file

File: tests/test_helpers.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
+from sqlalchemy.orm import Session
+
+from shout_subgroup.models import GroupChatModel, SubgroupModel, UserModel
+
+
+def create_test_user(session: Session, telegram_user_id, username, first_name, last_name) -> UserModel:
```
Comment (AshciR) [inline tests/test_helpers.py:6]: Refactor: Created 3 functions to assist with testing

File: tests/test_shout.py [hunk 7]
```diff
@@ -1,86 +1,93 @@
...
+    message = await shout_subgroup_members(db, group_chat.telegram_group_chat_id, subgroup_name)
 
     # Then: It mentions them
-    assert message == "@johndoe "
\ No newline at end of file
+    assert message == "@johndoe "
+
+
+@pytest.mark.asyncio
+async def test_shout_subgroup_members_only_mentions_members_for_group_chat(db: Session):
```
Comment (AshciR) [inline tests/test_shout.py:47]: Focus: Test case associated with this PR

**Review by rick1e — APPROVED** [review]

**Pull request merged**